openstudiocore/src/model/GeneratorPhotovoltaic.hpp

```diff
@@ -0,0 +1,143 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2015, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#ifndef MODEL_GENERATORPHOTOVOLTAIC_HPP
+#define MODEL_GENERATORPHOTOVOLTAIC_HPP
+
+#include <model/ModelAPI.hpp>
+#include "Generator.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+class PlanarSurface;
+class PhotovoltaicPerformance;
+class Schedule;
+
+namespace detail {
+
+  class GeneratorPhotovoltaic_Impl;
+
+} // detail
+
+/** GeneratorPhotovoltaic is a Generator that wraps the OpenStudio IDD object 'OS:Generator:Photovoltaic'. */
+class MODEL_API GeneratorPhotovoltaic : public Generator {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  /// Creates a GeneratorPhotovoltaic object with PhotovoltaicPerformanceSimple performance
+  static GeneratorPhotovoltaic simple(const Model& model);
```

Comment:
These factory methods are a little different pattern than other objects, they construct a required performance object of the right type.  Curious what @kbenne and others think of this?